mindsdb/interfaces/knowledge_base/controller.py

```diff
@@ -935,7 +935,16 @@ def add(
             model_record = db.Predictor.query.get(model["id"])
             embedding_model_id = model_record.id
 
-        reranking_model_params = get_model_params(params.get("reranking_model", {}), "default_reranking_model")
+        # if params.get("reranking_model", {}) is bool and False we evaluate it to empty dictionary
+        reranking_model_params = params.get("reranking_model", {})
+
+        if isinstance(reranking_model_params, bool) and not reranking_model_params:
+            params["reranking_model"] = {}
+        # if params.get("reranking_model", {}) is string and false in any case we evaluate it to empty dictionary
+        if isinstance(reranking_model_params, str) and reranking_model_params.lower() == "false":
+            params["reranking_model"] = {}
```

Comment:
it won't disable reranking if default values are in config (thay are added by get_model_params)
I can suggest:
- save it as `params["reranking_model"]=False`
- in `get_model_params` add condition:
  -  if model_params is False: return {}